src/Home/Home.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { map, curry } from "ramda";
+
+import FilterControls from "@/FilterControls/FilterControls"
+import MealCard from "@/MealCard/MealCard";
+import Overview from "@/Overview/Overview";
+
+
+const renderMealCards = (handler) => map((meal) => {
+	return <MealCard meal={ meal } key={ meal.id } handleUseIt={ handler.bind(null, meal.id) }/>
```

Comment:
One thing to note about the above - when you do: `handleuseIt={handler.bind(null, meal.id)}` - the `.bind()` may be called repeatedly, each time the `.render()` is invoked, creating a new function (identical to the old). Thus - you may be creating a *lot* of functions, over time - esp. since this is also being performed inside a `.map()`.

It's generally a good pattern to persist the bound functions and reuse them - or perhaps better yet, to bind more simply. Consider this pattern instead:

```javascript
class Parent extends React.Component {
  handleChildClick = (id) => {
    const child = this.props.collection.find(m => m.id === id)
    takeSomeActionOn(child)
  }
  render = () => (
    <div>
      {this.props.collection.map(childModel =>
        <Child id={childModel.id} onClick={this.handleChildClick} />
      )}
    </div>
  )
}

class Child extends React.Component {
  handleClick = () => { this.props.onClick(this.props.id) }
  render = () => <button onClick={this.handleClick} />
}
```
This can become more terse if using something like `MST` or `Redux`, where you can act directly from the child context - but that's not the point of the example. The point is - the child passes back its own identifier & other relevant information as part of raising the event, and thus the parent can respond to it without having to perform a `.bind()` as part of render.

Reply:
I had not thought of that. Great point.